contracts/features-staking/StakingCampaignRecipient.sol

```diff
@@ -0,0 +1,152 @@
+        address[] calldata _stakers,
+        uint256[] calldata _stakingAmounts,
+        uint256[][] calldata _rewardAmounts
+    ) external onlyOwner {
+        uint256 stakersLength = _stakers.length;
+        uint256 stakingAmountsLength = _stakingAmounts.length;
+
+        require(
+            stakersLength == stakingAmountsLength,
+            'StakingCampaignRecipient: stakers and staking amounts length mismatch'
+        );
+
+        for (uint256 i = 0; i < stakersLength; i++) {
+            _addStaker(_stakingAmounts[i], _stakers[i], _rewardAmounts[i]);
+        }
+    }
+
+    function _addStaker(
+        uint256 _tokenAmount,
+        address _staker,
+        uint256[] calldata _rewardAmount
+    ) internal {
+        uint256 currentTimestamp = block.timestamp;
+        require(
+            (startTimestamp > 0 && currentTimestamp > startTimestamp) &&
+                (currentTimestamp <= endTimestamp + extensionDuration),
+            'RewardsPoolBase: staking is not started or is finished or no extension taking in place'
+        );
+
+        UserInfo storage user = userInfo[_staker];
+        require(
+            (user.amountStaked + _tokenAmount <= stakeLimit) && (totalStaked + _tokenAmount <= contractStakeLimit),
+            'RewardsPoolBase: stake limit reached'
+        );
+
+        require(_tokenAmount > 0, 'RewardsPoolBase: cannot stake 0');
+
+        // if no amount has been staked this is considered the initial stake
+        if (user.amountStaked == 0) {
+            user.firstStakedTimestamp = currentTimestamp;
+        }
+
+        user.amountStaked = _tokenAmount;
+        pendingRewardsAmount[_staker] = _rewardAmount;
+        totalStaked = totalStaked + _tokenAmount;
+
+        updateRewardMultipliers();
+        _updateUserAccruedReward(_staker); // Update the accrued reward for this specific user
+
+        emit Staked(_staker, _tokenAmount);
+    }
+
+    function finalizeExit(address _stakingToken, address[] memory _rewardsTokens) internal override returns (uint256) {
+        ExitInfo storage info = exitInfo[msg.sender];
+        require(block.timestamp > info.exitTimestamp, 'finalizeExit::Trying to exit too early');
+
+        uint256 infoExitStake = info.exitStake;
+        require(infoExitStake > 0, 'finalizeExit::No stake to exit');
+        info.exitStake = 0;
+
+        IERC20(_stakingToken).safeTransfer(address(msg.sender), infoExitStake);
+
+        for (uint256 i = 0; i < _rewardsTokens.length; i++) {
+            uint256 infoRewards = info.rewards[i] + pendingRewardsAmount[msg.sender][i];
+            info.rewards[i] = 0;
+
+            IERC20(_rewardsTokens[i]).safeTransfer(msg.sender, infoRewards);
+        }
+
+        emit ExitCompleted(msg.sender, infoExitStake);
+
+        return infoExitStake;
+    }
+
+    function getUserAccumulatedReward(
+        address _userAddress,
+        uint256 _tokenIndex,
+        uint256 _time
+    ) external view override returns (uint256) {
```

Comment:
Inside this function you can just call the RewardsPoolBase.getUserAccumulatedReward and then add pendingRewards, no need to copy paste everything